x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/details/components/log_entry.tsx

```diff
@@ -6,52 +6,106 @@
  */
 
 import React, { memo } from 'react';
+import styled from 'styled-components';
 
-import { EuiAvatar, EuiComment, EuiText } from '@elastic/eui';
-import { Immutable, EndpointAction } from '../../../../../../../common/endpoint/types';
+import { EuiAvatar, EuiAvatarProps, EuiCommentProps, EuiComment, EuiText } from '@elastic/eui';
+import { Immutable, ActivityLogEntry } from '../../../../../../../common/endpoint/types';
 import { FormattedDateAndTime } from '../../../../../../common/components/endpoint/formatted_date_time';
 import { useEuiTheme } from '../../../../../../common/lib/theme/use_eui_theme';
 
-export const LogEntry = memo(
-  ({ endpointAction }: { endpointAction: Immutable<EndpointAction> }) => {
-    const euiTheme = useEuiTheme();
-    const isIsolated = endpointAction?.data.command === 'isolate';
+const StyledEuiComment = styled(EuiComment)`
+  .euiCommentEvent__headerTimestamp {
+    display: flex;
+    :before {
+      content: '';
+      background-color: ${(props) => props.theme.eui.euiColorInk};
+      display: block;
+      width: ${(props) => props.theme.eui.euiBorderWidthThick};
+      height: ${(props) => props.theme.eui.euiBorderWidthThick};
+      margin: 0 ${(props) => props.theme.eui.euiSizeXS} 0 ${(props) => props.theme.eui.euiSizeS};
+      border-radius: 50%;
+      content: '';
+      margin: 0 8px 0 4px;
+      border-radius: 50%;
+      position: relative;
+      top: 10px;
+    }
+  }
+`;
+export const LogEntry = memo(({ logEntry }: { logEntry: Immutable<ActivityLogEntry> }) => {
+  const euiTheme = useEuiTheme();
+  let iconType = 'dot';
```

Comment:
Can you move all of this outside the component? Maybe in some builder.ts with a function that receives a logEntry and returns an object with all of these values

Reply:
Yeah, so I go around to doing this anyway. 😄  6c7c3399c000548fff01716065dcd983dcaefed4